### ISSUE TITLE
False positive in options validator for valid config in declaration-block-properties-order

### ISSUE THREAD

**Issue by hudochenkov** [stylelint/stylelint#1948]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A bug with tests for very specific config in `declaration-block-properties-order` rule.

> Which rule, if any, is this issue related to?

`declaration-block-properties-order`

> What CSS is needed to reproduce this issue?

None.

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
    "declaration-block-properties-order": [
      "height",
      {
        "properties": [
          "color",
        ]
      }
    ]
  }
}
```

> Which version of stylelint are you using?

`7.3.1`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

`npm test`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No.

> What did you expect to happen?

Test passes.

> What actually happened (e.g. what warnings or errors you are getting)?

I found one edge case when options tests, which test rule options, are failing, but config is valid. If config consists with one property and following one object.

These warnings happens:

```
[ { text: 'Invalid option ""height"" for rule declaration-block-properties-order' },
  { text: 'Invalid option name "properties" for rule "declaration-block-properties-order"' } ]
```

If property goes after an object — no error. If there two properties before an object, or two objects after a property, also no error.

I've added test in PR https://github.com/stylelint/stylelint/pull/1949

**Comment by jeddy3** [comment]
@hudochenkov Thank you for the excellent report and for the tests!

**Comment by davidtheclark** [comment]
@hudochenkov three cheers for providing tests! I am going to make sure the tests you've provided pass as part of #1930.

**Comment by davidtheclark** [comment]
This is an unfortunate result of our imitation of ESLint using arrays for rule configuration instead of objects. A mistake we must live with now :(

There's no way to generically know that this config is not a primary option followed by secondary options. So we're going to have to make a very hacky rule-specific exception.

**Comment by jeddy3** [comment]
> This is an unfortunate result of our imitation of ESLint using arrays for rule configuration instead of objects. A mistake we must live with now :(

I hadn't considered that being the cause, but it does make sense now that you mention it.

> So we're going to have to make a very hacky rule-specific exception.

Agreed. :(

**Comment by hudochenkov** [comment]
> There's no way to generically know that this config is not a primary option followed by secondary options. So we're going to have to make a very hacky rule-specific exception.

I forgot to mention, that I [found this problem](https://github.com/hudochenkov/stylelint-declaration-block-order/blob/c081feb12be653db9d524498c68cbb95b9de0456/test/validate-options.js#L93) while been working on my plugin. This issue won't be fixed for plugins, I assume?

**Comment by davidtheclark** [comment]
@hudochenkov: I would inform your users that they *must* use a *nested* array for the primary option.

```json
[["something", { "foo": "bar" }]]
```

not

```json
["something", { "foo": "bar" }]
```

**Comment by davidtheclark** [comment]
Fixed in #1930.
